mbuild/compound.py

```diff
@@ -523,6 +524,120 @@ def update_coordinates(self, filename):
+               MMFF94 using experimental data, additional computational data,
+               and empirical rules." (1996) J. Comput. Chem. 17, 616-641
+
+        If using the 'UFF' force field please also cite the following:
+        .. [3] Rappe, A.K., Casewit, C.J., Colwell, K.S., Goddard, W.A. III,
+               Skiff, W.M. "UFF, a full periodic table force field for
+               molecular mechanics and molecular dynamics simulations." (1992)
+               J. Am. Chem. Soc. 114, 10024-10039
+
+        If using the 'GAFF' force field please also cite the following:
+        .. [3] Wang, J., Wolf, R.M., Caldwell, J.W., Kollman, P.A., Case, D.A.
+               "Development and testing of a general AMBER force field" (2004)
+               J. Comput. Chem. 25 1157-1174
+        """
+        try:
+            import openbabel
+        except ImportError:
+            warn("Open Babel package not detected. Energy minimization will not "
+                 "be performed.", ImportWarning)
+            return
+
+        tmp_dir = tempfile.mkdtemp()
+        original = clone(self)
+        self.tweak()
+        self.save(os.path.join(tmp_dir,'un-minimized.pdb'))
+        obConversion = openbabel.OBConversion()
+        obConversion.SetInAndOutFormats("pdb", "mol2")
+        mol = openbabel.OBMol()
+
+        obConversion.ReadFile(mol, os.path.join(tmp_dir, "un-minimized.pdb"))
+
+        if forcefield == 'MMFF94':
+            ff = openbabel.OBForceField.FindForceField("MMFF94")
+        elif forcefield == 'UFF':
+            ff = openbabel.OBForceField.FindForceField("UFF")
+        elif forcefield == 'GAFF':
+            ff = openbabel.OBForceField.FindForceField("GAFF")
+        else:
+            forcefield = 'UFF'
+            ff = openbabel.OBForceField.FindForceField("UFF")
+            warn("Invalid forcefield (valid options are 'MMFF94', 'UFF', and "
+                 "'GAFF'). Using UFF.", RuntimeWarning)
+        warn("Performing energy minimization using the Open Babel package. Please "
+             "refer to the documentation to find the appropriate citations for "
+             "Open Babel and the {} force field".format(forcefield))
+        ff.Setup(mol)
+        if algorithm == 'steep':
+            ff.SteepestDescent(steps)
+        elif algorithm == 'cg':
+            ff.ConjugateGradients(steps)
+        else:
+            ff.ConjugateGradients(steps)
+            warn("Invalid minimization algorithm (valid options are 'steep' "
+                 "and 'cg'). Using conjugate gradient algorithm.", RuntimeWarning)
+            ff.UpdateCoordinates(mol)
+
+        obConversion.WriteFile(mol, os.path.join(tmp_dir, 'minimized.mol2'))
+        try:
+            self.update_coordinates(os.path.join(tmp_dir, 'minimized.mol2'))
+        except ValueError:
+            for i, particle in enumerate(self.particles()):
+                particle.pos = original.xyz[i]
```

Comment:
Yes.  This way if something fails along the way with the optimization the Compound will remain unchanged from its state prior to attempting energy minimization.